File: bench/python/wordcount.py
import os
N = int(os.environ.get("BENCH_N", "100000"))
K = 1000

x = 123456789
counts = {}
for _ in range(N):
    x = (x * 1103515245 + 12345) & 0x7FFFFFFF
    key = x % K
    counts[key] = counts.get(key, 0) + 1

total = 0
for k, v in counts.items():
    total += k * v

print(total)

File: bench/go/wordcount.go
// LCG token stream tallied into a map — the plain read-modify-write, `counts[key]++`.
package main

import "fmt"

func main() {
	n := benchN(750000)
	const k = 1000
	x := 123456789
	counts := make(map[int]int)
	for i := 0; i < n; i++ {
		x = (x*1103515245 + 12345) & 0x7FFFFFFF
		counts[x%k]++
	}
	total := 0
	for key, v := range counts {
		total += key * v
	}
	fmt.Println(total)
}
